src/Features/CSharp/Portable/ReplacePropertyWithMethods/CSharpReplacePropertyWithMethodsService.cs

```diff
@@ -0,0 +1,182 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Composition;
+using System.Linq;
+using System.Threading;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CSharp.Extensions;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.ReplacePropertyWithMethods;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.ReplacePropertyWithMethods
+{
+    [ExportLanguageService(typeof(IReplacePropertyWithMethodsService), LanguageNames.CSharp), Shared]
+    internal class CSharpReplacePropertyWithMethodsService : 
+        AbstractReplacePropertyWithMethodsService<IdentifierNameSyntax, ExpressionSyntax, StatementSyntax>
+    {
+        public override SyntaxNode GetPropertyDeclaration(SyntaxToken token)
+        {
+            var containingProperty = token.Parent.FirstAncestorOrSelf<PropertyDeclarationSyntax>();
+            if (containingProperty == null)
+            {
+                return null;
+            }
+
+            var start = containingProperty.AttributeLists.Count > 0
+                ? containingProperty.AttributeLists.Last().GetLastToken().GetNextToken().SpanStart
+                : containingProperty.SpanStart;
+
+            // Offer this refactoring anywhere in the signature of the method.
```

Comment:
It seems to be the same.  ChangeSignature handles a lot of types of VB statements. However, for the property case, these seem to match.